### ISSUE TITLE
getCredentialWith method misbehaves when closing the browser with the "End button" or following an authentication error

### ISSUE THREAD

**Issue by dennysoft** [firebase/firebase-ios-sdk#13645]
### Description

I have updated the Firebase SDK to version 11.2.0. Using the get Credential With method to connect to a provider using oidc connect. If an error occurs during authentication or I close the browser with the Finish button, the method receives the error (browser closed by user error code: 17058) but the app controller is closed it seems to lose context and app window closes I have no possibility to show the error on the screen. This was not the case with previous versions of the Firebase SDK. It's not a bug in my app as the same code with version 10.27.0 works perfectly.

### Reproducing the issue

I instantiate the OAuthProvider object and launch the Get Credential With method with a button, the browser opens and I press the finish button. The app window closes.

### Firebase SDK Version

11.2.0

### Xcode Version

15.4

### Installation Method

Swift Package Manager

### Firebase Product(s)

Authentication

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by ncooke3** [comment]
Thanks for the report, @dennysoft, and apologies for the trouble. I have a couple questions to help me better understand the behavior you are seeing.

- Do you have swizzling enabled or disabled? If you haven't included `FirebaseAppDelegateProxyEnabled: NO`, then it's enabled.
- Are you able to provide screenshots of the difference in behavior you are seeing?

**Comment by dennysoft** [comment]
Hi @ncooke3

- i haven't included  FirebaseAppDelegateProxyEnabled: NO
- when call getcredentialwith and then close browser with end button (not complete authentication) the app gets restarted 
- i use Appdelegate with  func application( _ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:] ) -> Bool

**Comment by ncooke3** [comment]
Hi @dennysoft, we've identified the issue and will have a fix (#13715) tentatively scheduled to be released next week.

**Comment by dennysoft** [comment]
Thanks!
